WordPress/WordPressTest/TestUtilities/JSONObject.swift

```diff
@@ -7,23 +7,34 @@ extension JSONObject {
 
     /// Loads the specified json file and returns a dictionary representing it.
     ///
-    /// - Parameter fileName: The full name of the json file to load.
+    /// - Parameter fileName: The name of the json file to load. The "json" file extension can be omitted.
     /// - Returns: A dictionary representing the contents of the json file.
-    static func loadFile(named fileName: String) throws -> JSONObject {
-        return try loadFile(
-            (fileName as NSString).deletingPathExtension,
-            type: (fileName as NSString).pathExtension
-        )
-    }
+    static func loadJSONFile(named fileName: String) throws -> JSONObject {
+        let type = (fileName as NSString).pathExtension
+        if type != "" && type != "json" {
+            throw NSError(
+                domain: "JSONObject",
+                code: 1,
+                userInfo: [
+                    NSLocalizedDescriptionKey: "File extension should be 'json': \(fileName)"
+                ]
+            )
```

Comment:
Why an `NSError` vs something simpler that conforms to `Error`?

Reply:
Also, in this case it looks like we're defending against programmer error (ie – entering a malformed filename), so I wonder if using `assert` would work?

Reply:
I didn't create a new `Error` type here because I don't think a new type has much value in this context. I don't think the errors thrown in this function will be catched and handled in any place, like you said they are programmer errors. IMHO, they are intended to be used as a "test failure" which shows up in the test result, to avoid crashing the whole unit test suite like [the previous implementation with a fatalError](https://github.com/wordpress-mobile/WordPress-iOS/pull/18507), when an issue does occur.

Reply:
> I don't think a new type has much value in this context

> they are intended to be used as a "test failure" which shows up in the test result, to avoid crashing the whole unit test suite

Makes sense – thanks for explaining!